test/test.cpp

```diff
@@ -3,10 +3,378 @@ bundle config --local path vendor/bundle
+//     delete pin;
+//     pin = nullptr;
+//     delay(11);
+//     keypad.getKey();
+//     delay(11);
+//     assertEqual((int)'\0', (int)keypad.getKey());
+
+//     pin = new BitCollector(8, 4);
+//     delay(100);
+//     for (int i = 2; i <= 5; ++i)
+//     {
+//         state->digitalPin[i] = HIGH;
+//     }
+//     assertEqual((int)'5', (int)keypad.getKey());
+//     delete pin;
+//     pin = nullptr;
+//     delay(11);
+//     keypad.getKey();
+//     delay(11);
+//     assertEqual((int)'\0', (int)keypad.getKey());
+
+//     pin = new BitCollector(8, 3);
+//     delay(100);
+//     for (int i = 2; i <= 5; ++i)
+//     {
+//         state->digitalPin[i] = HIGH;
+//     }
+//     assertEqual((int)'8', (int)keypad.getKey());
+//     delete pin;
+//     pin = nullptr;
+//     delay(11);
+//     keypad.getKey();
+//     delay(11);
+//     assertEqual((int)'\0', (int)keypad.getKey());
+
+//     pin = new BitCollector(8, 2);
+//     delay(100);
+//     for (int i = 2; i <= 5; ++i)
+//     {
+//         state->digitalPin[i] = HIGH;
+//     }
+//     assertEqual((int)'0', (int)keypad.getKey());
+//     delete pin;
+//     pin = nullptr;
+//     delay(11);
+//     keypad.getKey();
+//     delay(11);
+//     assertEqual((int)'\0', (int)keypad.getKey());
+
+//     pin = new BitCollector(9, 5);
+//     delay(100);
+//     for (int i = 2; i <= 5; ++i)
+//     {
+//         state->digitalPin[i] = HIGH;
+//     }
+//     assertEqual((int)'1', (int)keypad.getKey());
+//     delete pin;
+//     pin = nullptr;
+//     delay(11);
+//     keypad.getKey();
+//     delay(11);
+//     assertEqual((int)'\0', (int)keypad.getKey());
+
+//     pin = new BitCollector(9, 4);
+//     delay(100);
+//     for (int i = 2; i <= 5; ++i)
+//     {
+//         state->digitalPin[i] = HIGH;
+//     }
+//     assertEqual((int)'4', (int)keypad.getKey());
+//     delete pin;
+//     pin = nullptr;
+//     delay(11);
+//     keypad.getKey();
+//     delay(11);
+//     assertEqual((int)'\0', (int)keypad.getKey());
+
+//     pin = new BitCollector(9, 3);
+//     delay(100);
+//     for (int i = 2; i <= 5; ++i)
+//     {
+//         state->digitalPin[i] = HIGH;
+//     }
+//     assertEqual((int)'7', (int)keypad.getKey());
+//     delete pin;
+//     pin = nullptr;
+//     delay(11);
+//     keypad.getKey();
+//     delay(11);
+//     assertEqual((int)'\0', (int)keypad.getKey());
+
+//     pin = new BitCollector(9, 2);
+//     delay(100);
+//     for (int i = 2; i <= 5; ++i)
+//     {
+//         state->digitalPin[i] = HIGH;
+//     }
+//     assertEqual((int)'*', (int)keypad.getKey());
+//     delete pin;
+//     pin = nullptr;
+//     delay(11);
+//     keypad.getKey();
+//     delay(11);
+//     assertEqual((int)'\0', (int)keypad.getKey());
+// }
+
+unittest(highlevel) 
+{
+    GodmodeState *state;
+    state = GODMODE();
+    for (int i = 2; i <= 5; ++i)
+    {
+        std::cout << i << " " << (int)(state->digitalPin[i]) << std::endl;
```

Comment:
If this was for debugging and things now work it should probably come out. Generally an automated test should pass (mostly) silently. Output should be reserved for test failures.